README.md
<div id="table-of-contents">
<h2>Table of Contents</h2>
<div id="text-table-of-contents">
<ul>
<li><a href="#sec-1">1. Packages</a>
<ul>
<li><a href="#sec-1-1">1.1. Cask</a></li>
<li><a href="#sec-1-2">1.2. Pallet</a></li>
<li><a href="#sec-1-3">1.3. Package.el</a></li>
<li><a href="#sec-1-4">1.4. Paradox</a></li>
</ul>
</li>
<li><a href="#sec-2">2. Uniquify</a></li>
<li><a href="#sec-3">3. Ido</a></li>
<li><a href="#sec-4">4. Evil</a></li>
<li><a href="#sec-5">5. Line Numbers</a></li>
<li><a href="#sec-6">6. Winner-mode</a></li>
<li><a href="#sec-7">7. pbcopy</a></li>
<li><a href="#sec-8">8. Programming</a>
<ul>
<li><a href="#sec-8-1">8.1. General</a></li>
<li><a href="#sec-8-2">8.2. Git</a></li>
<li><a href="#sec-8-3">8.3. Projectile</a></li>
<li><a href="#sec-8-4">8.4. Paredit</a></li>
<li><a href="#sec-8-5">8.5. Emacs Lisp</a></li>
<li><a href="#sec-8-6">8.6. Javascript</a></li>
<li><a href="#sec-8-7">8.7. PHP</a></li>
</ul>
</li>
<li><a href="#sec-9">9. Compilation</a></li>
<li><a href="#sec-10">10. Org-mode</a></li>
<li><a href="#sec-11">11. IRC</a></li>
<li><a href="#sec-12">12. WWW</a></li>
<li><a href="#sec-13">13. Email</a></li>
<li><a href="#sec-14">14. Terminals</a></li>
<li><a href="#sec-15">15. Man Pages</a></li>
<li><a href="#sec-16">16. Miscellaneous Functions</a></li>
<li><a href="#sec-17">17. Regular Expressions</a></li>
<li><a href="#sec-18">18. Key Bindings</a></li>
<li><a href="#sec-19">19. Theme</a></li>
<li><a href="#sec-20">20. Backup and Auto Save</a></li>
<li><a href="#sec-21">21. General Setup</a></li>
<li><a href="#sec-22">22. Local Setup</a></li>
</ul>
</div>
</div>


# Packages

## Cask

Cask is a dependency manager for Emacs, similar to npm for Node.

[Cask website](http://cask.github.io)

First, we require and initialize Cask.

    (require 'cask "~/.cask/cask.el")
    (cask-initialize)

## Pallet

Pallet keeps the Cask file up-to-date when installing packages through
\`M-x list-packages\`.

[Pallet website](https://github.com/rdallasgray/pallet)

    (require 'pallet)

## Package.el

Also, we need to set up the package repositories for Emacs' own package
manager.

    (setq package-archives
          '(("gnu" . "http://elpa.gnu.org/packages/")
            ("marmalade" . "http://marmalade-repo.org/packages/")
            ("melpa" . "http://melpa.milkbox.net/packages/")))

## Paradox

Finally, paradox is a thing.

    (setq paradox-github-token t)

# Uniquify

If I have two buffers open with two files that have the same name, (e.g. two 
different README files from two different projects), Emacs will, by default, 
name the buffers \`README\` and \`README<1>\`. This is useless. Therefore, I turn on 
uniquify and use it to name buffers wtih the same file name based on their 
parent directories: \`README<projdir1>\` and \`README<projectdir2>\`.

    (require 'uniquify)
    (setq uniquify-buffer-name-style 'post-forward-angle-brackets)

# Ido

Ido makes minibuffer completions much nicer, easier, and faster.

    (require 'ido)
    (ido-mode)
    (setq ido-completion-buffer "*Ido Completions*")
    (setq ido-completion-buffer-all-completions t)
    (setq ido-enable-flex-matching t)
    (setq ido-everywhere t)
    (icomplete-mode 99)
    (ido-vertical-mode 1)

# Evil

Call me heathen if you wish, but I prefer Vim navigation keys.

[Evil website](https://gitorious.org/evil/pages/Home)

    (evil-mode 1)
    (setq evil-default-cursor '(t))

Using \`key-chord-mode\`, I have the vim equivalent of \`imap jk <Esc>\`, which 
allows me to to \`jk\` instead of \`Esc\` to get out of insert mode.

    (key-chord-mode 1)
    (key-chord-define evil-insert-state-map "jk" 'evil-normal-state)
    (key-chord-define evil-motion-state-map "jk" 'evil-normal-state)
    (key-chord-define evil-visual-state-map "jk" 'evil-normal-state)
    (key-chord-define evil-emacs-state-map "jk" 'evil-normal-state)

I want Ido buffer switching and file finding when using Vim's \`:b\` and \`:e\`.

    (define-key evil-ex-map "b " 'ido-switch-buffer)
    (define-key evil-ex-map "e " 'ido-find-file)

I also want Vim's \`vs\` command for surrounding text with quotes, braces, etc.

    (require 'surround)
    (global-surround-mode 1)

To make things even easier, I set up a "leader key" of \`Space\`, so that I can
type \`Space <letter>\` to run a command. For instance, \`Space x\` instead of
\`M-x\` to execute commands.

    (require 'evil-leader)
    (evil-leader/set-leader "<SPC>")
    (evil-leader/set-key "x" 'execute-extended-command)
    (evil-leader/set-key ":" 'eval-expression)
    (evil-leader/set-key "k" 'ido-kill-buffer)
    (evil-leader/set-key "p" 'projectile-commander)
    (evil-leader/set-key "e" 'mu4e)
    (evil-leader/set-key "j" 'ace-jump-mode)
    (evil-leader/set-key "J" 'ace-window)
    (global-evil-leader-mode)

Finally, there are some modes that I want to always be in Emacs mode instead
of Evil.

    (setq evil-emacs-state-modes
          '(archive-mode bbdb-mode bookmark-bmenu-mode bookmark-edit-annotation-mode browse-kill-ring-mode bzr-annotate-mode calc-mode cfw:calendar-mode completion-list-mode Custom-mode debugger-mode delicious-search-mode desktop-menu-blist-mode desktop-menu-mode doc-view-mode dvc-bookmarks-mode dvc-diff-mode dvc-info-buffer-mode dvc-log-buffer-mode dvc-revlist-mode dvc-revlog-mode dvc-status-mode dvc-tips-mode ediff-mode ediff-meta-mode efs-mode Electric-buffer-menu-mode emms-browser-mode emms-mark-mode emms-metaplaylist-mode emms-playlist-mode etags-select-mode fj-mode gc-issues-mode gdb-breakpoints-mode gdb-disassembly-mode gdb-frames-mode gdb-locals-mode gdb-memory-mode gdb-registers-mode gdb-threads-mode gist-list-mode gnus-article-mode gnus-browse-mode gnus-group-mode gnus-server-mode gnus-summary-mode google-maps-static-mode ibuffer-mode jde-javadoc-checker-report-mode magit-commit-mode magit-diff-mode magit-key-mode magit-log-mode magit-mode magit-reflog-mode magit-show-branches-mode magit-branch-manager-mode magit-stash-mode magit-status-mode magit-wazzup-mode mh-folder-mode monky-mode mu4e-main-mode mu4e-headers-mode mu4e-view-mode notmuch-hello-mode notmuch-search-mode notmuch-show-mode occur-mode org-agenda-mode package-menu-mode proced-mode rcirc-mode rebase-mode recentf-dialog-mode reftex-select-bib-mode reftex-select-label-mode reftex-toc-mode sldb-mode slime-inspector-mode slime-thread-control-mode slime-xref-mode sr-buttons-mode sr-mode sr-tree-mode sr-virtual-mode tar-mode tetris-mode tla-annotate-mode tla-archive-list-mode tla-bconfig-mode tla-bookmarks-mode tla-branch-list-mode tla-browse-mode tla-category-list-mode tla-changelog-mode tla-follow-symlinks-mode tla-inventory-file-mode tla-inventory-mode tla-lint-mode tla-logs-mode tla-revision-list-mode tla-revlog-mode tla-tree-lint-mode tla-version-list-mode twittering-mode urlview-mode vc-annotate-mode vc-dir-mode vc-git-log-view-mode vc-svn-log-view-mode vm-mode vm-summary-mode w3m-mode wab-compilation-mode xgit-annotate-mode xgit-changelog-mode xgit-diff-mode xgit-revlog-mode xhg-annotate-mode xhg-log-mode xhg-mode xhg-mq-mode xhg-mq-sub-mode xhg-status-extra-mode cider-repl-mode emacsagist-mode elfeed-show-mode elfeed-search-mode notmuch-tree term-mode))

# Line Numbers

I like line numbers. They help quite a bit with moving around. Here
I turn on linum-mode globally.

    (global-linum-mode)

Then I change the way line numbers are
displayed to be right-justified.

    (setq linum-format (lambda
                         (line)
                         (propertize
                          (format (concat "%"
                                          (number-to-string
                                           (length
                                            (number-to-string
                                             (line-number-at-pos
                                              (point-max)))))
                                          "d ")
                                  line)
                          'face
                          'linum)))

# Winner-mode

Winner-mode makes it really easy to handle window changes in
Emacs. `C-c left-arrow` goes back to the previous window
configuration (undo), and `C-c right-arrow` goes forward
(redo). This is especially helpful for when a popop window ruins
your layout. Simply `C-c left-arrow` to get back to where you were.

    (winner-mode 1)

# pbcopy

Clipboard sharing. Copy in Emacs, paste in OS X, and vice versa.

[pbcopy source](https://github.com/jkp/pbcopy.el)

    (require 'pbcopy)
    (turn-on-pbcopy)

# Programming

## General

Here are some general, all-purpose, language-agnostic default settings.

    (setq c-basic-offset 4)
    (setq tab-width 4)

## Git

Magit is awesome.

    (setq magit-emacsclient-executable "/usr/local/bin/emacsclient")
    (setq magit-server-window-for-commit 'pop-to-buffer)
    (setq magit-use-overlays nil)

## Projectile

Projectile is, quite simply and objectively, the shit. There's no other way to
put it.

[Projectile on Github](https://github.com/bbatsov/projectile)

    (projectile-global-mode)

## Paredit

If you write any form of Lisp and don't use paredit, change that. 

[Paredit website](http://mumble.net/~campbell/emacs/paredit.el)

[Emacs Rocks episode on paredit](http://emacsrocks.com/e14.html)

    (require 'paredit)
    (add-hook 'emacs-lisp-mode-hook 'paredit-mode)
    (add-hook 'clojure-mode-hook 'paredit-mode)

## Emacs Lisp

    (defun my-emacs-lisp-mode-hook ()
      (eldoc-mode)
      (linum-mode)
      (lexbind-mode))
    (add-hook 'emacs-lisp-mode-hook 'my-emacs-lisp-mode-hook)
    (add-hook 'clojure-mode-hook 'linum-mode)

## Javascript

Prefer js2-mode to javascript-mode.

    (autoload 'js2-mode "js2" nil t)
    (add-to-list 'auto-mode-alist '("\\.js$" . js2-mode))

## PHP

    (defun my-php-mode-hook ()
      (make-local-variable (quote whitespace-style))
      (setf whitespace-style (quote (face lines-tail tab-mark)))
      (whitespace-mode t)
      (linum-mode 1)
      (subword-mode 1)
      (php-enable-symfony2-coding-style))
    (add-hook 'php-mode-hook 'my-php-mode-hook)

# Compilation

\`M-x compile\` has some issues with ansi color codes. This fixes

elim/CHANGES
2010-04-13  Vivek Dasmohapatra  <vivek@etla.org>

	* elisp/garak.el:
	(garak-mapbuffers): apply action to all garak buffers
	(garak-buffer-account-matches): helper to find buffers for given account
	(garak-network-down): handle network down event
	(garak-disconnected): handle account disconnected event
	(garak-network-up): handle network up event (reconnect accounts)
	(garak-buddy-list-skip): better offline-contact check, buddies that go
	offline now disappear if offline buddy hiding is ocnfigured
	(garak-summarise-text): text summarisation (mainly for alerts)
	(garak-alert-user): put content of message into alert

	* elisp/elim.el (elim-parse-proto-args): allow empty arglist

	* elisp/dbus-util.el:
	(dbus-util-method-args-in): handle unnamed args
	(dbus-util-argspec-to-cons): ibid
	(dbus-util-call-method): allow elisp-style-names for dbus functions
	(dbus-util-deuglify-name): utilities to de-and-re uglify dbus names
	(dbus-util-deuglify-char): ibid
	(dbus-util-uglify-name): ibid
	(dbus-util-uglify-char): ibid

2010-03-26  Vivek Dasmohapatra  <vivek@etla.org>

	* elisp/garak.el:
	(garak-update-buddy): don't lose the proto icon from widgets on update

	* elisp/notify.el: org.freedesktop.Notifications interface

	* elisp/dbus-util.el: utility library for calling dbus methods

	* elisp/garak.el:
	(garak-alert-when): defcustom to pick when to alert the user
	(garak-alert-methods): defcustom to pick how to issue alerts
	(garak-recent-alerts): variable to track and throttle alerts
	(garak-alert-user): handle actual logic of alerting the user
	(garak-chat-message): hook up the alert function to the message handler

2009-07-03  Vivek Dasmohapatra  <vivek@etla.org>

	* handlers/set_icon.c (_h_elim_set_icon): silence a warning.

2009-07-02  Vivek Dasmohapatra  <vivek@etla.org>

	* sexp/sexp-util.h (ALIST_VAL_DATA): new alias for arg value macro

	* handlers/set_icon.c (_h_elim_set_icon): implement setting account icon

	* elisp/elim.el (elim-set-account-icon): call set-icon handler in daemon

	* elisp/garak.el:
	(garak-comp-account-file): completer for set-icon
	(garak-cmd-set-icon): CLI handler for /set-icon
	(garak-command-handlers): register CLI handler for /set-icon
	(garak-command-match): regex for same
	(garak-command-completers): register tab completer for /set-icon

2009-06-21  Vivek Dasmohapatra  <vivek@etla.org>

	* elisp/garak.el: 
	(garak-update-buddy): update label when buddy alias/name changes.
	(garak-cmd-alias-buddy): cli command interpreter for /alias
	(garak-command-handlers): register /alias handler
	(garak-command-match): /alias command regexp
	(garak-commands): /alias command added to list
	(garak-command-completers): register tab completion handler for /alias

	* elisp/elim.el (elim-alias-bnode): support daemon buddy node aliasing.

2009-06-20  Vivek Dasmohapatra  <vivek@etla.org>

	* elisp/garak.el (garak-comp-msg): tab completion of buddies for /msg

	* elisp/elim.el (elim-get-buddies): add util func to get buddies
	based on user-defined filters.

	* handlers/alias_bnode.h (_h_elim_alias_bnode): buddy aliasing support.
	* handlers/alias_bnode.c (_h_elim_alias_bnode): buddy aliasing support.

2009-06-19  Vivek Dasmohapatra  <vivek@etla.org>

	* elisp/garak.el:
	(garak-buddy-list-node-widget): prefer: alias > contact alias > name
	(garak-request-add-buddy): implement add buddy/cancel actions
	reaulting from elim-request-add-buddy calls.

2009-06-18  Vivek Dasmohapatra  <vivek@etla.org>

	* handlers/notify_search_callback.c (_h_elim_notify_search_callback): 
	use the cached row data rather than the ephemeral row data which may 
	have been freed by now ( judging by the example in the pidgin source, 
	this is the right approach here ).
	* ui_ops/notify_ui_ops.c (_elim_notify_search): call the standard purple
	`more search rows' function instead of assuming our callback is in place
	and calling it directly. Unlikely that we were wrong, but just in case.
	(_elim_notify_search_rows): cache the ephemeral row data we get here
	* elim-rpc.h (_NOTIFY_RESP): search row data is ephemeral: cache it here

2009-06-17  Vivek Dasmohapatra  <vivek@etla.org>

	* elisp/garak.el:
	(garak-callbacks): elim-blist-request-add-buddy handler registered
	(garak-request-add-buddy): handler for above elim call.

2009-06-14  Vivek Dasmohapatra  <vivek@etla.org>

	* handlers/notify_search_callback.c: 
	(_h_elim_notify_search_callback): check gc, row & button are all valid
	(_h_elim_notify_search_callback): get search cache with sid, not call id
	(_h_elim_notify_search_callback): extract row data correctly

	* elisp/garak.el (garak-search-results-buffer): cache args in buffer
	(garak-notify-search-callback): pass search id back in callback

	* elim-rpc.c (check_cb_data): abort if no key passed in

2009-06-13  Vivek Dasmohapatra  <vivek@etla.org>

	* elisp/garak.el: implement search reulst display/action UI.
	* ui_ops/notify_ui_ops.c (_elim_notify_search): pass on the account uid

2009-06-05  Vivek Dasmohapatra  <vivek@etla.org>

	* handlers/notify_search_callback.c: search callbacks (add, join etc)

custom/work.el
;-*- mode: elisp -*-
(custom-set-variables
 ;; custom-set-variables was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 '(auth-sources (quote ((:source "~/.authinfo.gpg" :host t :protocol nil))))
 '(auto-save-file-name-transforms (quote ((".*" "~/.emacs.tmp/" nil) ("\\`/[^/]*:\\([^/]*/\\)*\\([^/]*\\)\\'" "/var/folders/vO/vOsSP+j+Fxyyk9m8oLzG1E+++TM/-Tmp-/\\2" t))))
 '(auto-save-list-file-prefix "~/.emacs.tmp/.saves-")
 '(backup-directory-alist (quote (("" . "~/.emacs.tmp"))))
 '(blink-cursor-mode t)
 '(c-basic-offset 4)
 '(case-fold-search t)
 '(case-replace t)
 '(company-backends (quote (company-elisp company-nxml company-css company-eclim company-semantic company-xcode (company-etags company-dabbrev-code company-keywords php-completion) company-oddmuse company-files company-dabbrev)))
 '(company-begin-commands (quote (self-insert-command)))
 '(company-idle-delay nil)
 '(company-show-numbers t)
 '(compilation-error-regexp-alist nil)
 '(custom-theme-load-path (quote (custom-theme-directory t)))
 '(display-buffer-reuse-frames t)
 '(display-time-24hr-format nil)
 '(display-time-day-and-date t)
 '(emacs-setup-base-sexp (quote ((set-cursor-color "Green") (when (fboundp (quote toggle-scroll-bar)) (toggle-scroll-bar nil)) (global-font-lock-mode t) (transient-mark-mode t) (show-paren-mode t) (column-number-mode t) (when (fboundp (quote mouse-wheel-mode)) (mouse-wheel-mode t)) (temp-buffer-resize-mode 0) (add-hook (quote before-save-hook) (quote time-stamp)) (add-hook (quote after-make-frame-functions) (quote emacs-setup-base) t) (put (quote dired-find-alternate-file) (quote disabled) nil))))
 '(emacs-setup-elisp-base-dir "~/Dropbox/emacs")
 '(emacs-setup-elisp-ignore-dirs (quote (".svn")))
 '(emacs-setup-elpa-package-file "")
 '(emacs-setup-env-path-list (quote ("/opt/local/lib/mysql5/bin" "/opt/local/bin" "/usr/local/bin")))
 '(emacs-setup-frame-configuration (quote ("testing" "1920-1200-82-82-24-4.el")) t)
 '(emacs-setup-frame-configurations (quote (("test" "1920-1200-80-35-42-534.el" "1920-1200-82-76-22-1272.el"))))
 '(emacs-setup-frame-height 99)
 '(emacs-setup-frame-resume-window-layout nil)
 '(emacs-setup-frame-width 358)
 '(emacs-setup-frames nil)
 '(emacs-setup-fullscreen t)
 '(emacs-setup-keybindings (quote (("capitalize-word" . "C-c c") ("comment-region" . "C-c k") ("uncomment-region" . "C-c u") ("smex" . "M-x") ("scroll-up-line" . "M-n") ("scroll-down-line" . "M-p") ("other-window-backwards" . "C-x O") ("delete-other-windows-vertically" . "C-x 9") ("hs-goto-line" . "s-l") ("emms-previous" . "<f7>") ("emms-pause" . "<f8>") ("emms-next" . "<f9>") ("ido-switch-buffer" . "s-b"))))
 '(emacs-setup-layout-buffer-list (quote (("1920-1200-270-82-22-4.el" "*scratch*" "*Messages*" "*compilation*") ("1920-1200-82-82-24-4.el" "emacs-setup-layout.el") ("1920-1200-82-76-24-4.el" "emacs-setup-layout.el") ("1920-1200-80-35-25-652.el" "*compilation*") ("1920-1200-82-82-22-1272.el" "*Messages*" "*Help*") ("1920-1200-80-35-42-534.el" "*Messages*") ("1920-1200-82-76-22-1272.el" "*scratch*" "*compilation*") ("1920-1200-268-85-0-0.el" "*scratch*" "*Messages*" "*compilation*") ("2560-1440-358-99-22-4.el" "*scratch*" "*scratch*" "*Messages*" "*compilation*") ("2560-1440-360-102-0-0.el" "*scratch*" "*scratch*" "*Messages*" "*compilation*") ("425-101-425-101-0-0.el" "*Messages*" "*scratch*" "*scratch*" "*scratch*" "*scratch*") ("80-23-80-23-0-0.el" "*scratch*") ("425-99-425-99-0-0.el" "*scratch*" "*invalid-packages*" "*invalid-packages*" "*invalid-packages*" "*invalid-packages*") ("170-67-170-67-0-0.el" "*Messages*" "*scratch*"))))
 '(emacs-setup-layout-dir "~/Dropbox/emacs/custom/layouts")
 '(emacs-setup-load-elpa t)
 '(emacs-setup-load-layout t)
 '(emacs-setup-load-path-list nil)
 '(emacs-setup-load-window-number-mode t)
 '(emacs-setup-post-sexp (quote ((when (fboundp (quote rainbow-delimiters-mode)) (add-hook (quote emacs-lisp-mode-hook) (quote rainbow-delimiters-mode))) (when (fboundp (quote paredit-mode)) (add-hook (quote emacs-lisp-mode-hook) (quote paredit-mode))) (setq completion-ignore-case t) (autoload (quote garak) "garak" nil t))))
 '(emacs-setup-pre-layout-sexp nil)
 '(emacs-setup-pre-sexp (quote (nil)))
 '(emacs-setup-require-list (quote (("uniquify") ("ido" (ido-mode) (icomplete-mode 99)) ("smex-autoloads") ("evil" (evil-mode) (define-key evil-ex-map "e " (quote ido-find-file)) (define-key evil-ex-map "b " (quote ido-switch-buffer)) (define-key evil-window-map "" (quote evil-window-next))) ("my-funcs") ("zf-mode") ("emms-setup" (emms-standard) (emms-default-players)) ("gnus-notify"))))
 '(emacs-setup-set-frame-size nil)
 '(emms-player-list (quote (emms-player-mplayer-playlist emms-player-mplayer emms-player-vlc)))
 '(emms-player-mplayer-command-name "/usr/local/bin/mplayer")
 '(emms-source-file-default-directory "~/Music/iTunes/iTunes Media/Music")
 '(erc-fill-column 70)
 '(eshell-post-command-hook nil)
 '(eudc-strict-return-matches nil)
 '(evil-cross-lines nil)
 '(evil-default-state (quote normal))
 '(evil-emacs-state-modes (quote (archive-mode bbdb-mode bookmark-bmenu-mode bookmark-edit-annotation-mode browse-kill-ring-mode bzr-annotate-mode calc-mode cfw:calendar-mode completion-list-mode debugger-mode delicious-search-mode desktop-menu-blist-mode desktop-menu-mode doc-view-mode dvc-bookmarks-mode dvc-diff-mode dvc-info-buffer-mode dvc-log-buffer-mode dvc-revlist-mode dvc-revlog-mode dvc-status-mode dvc-tips-mode ediff-mode efs-mode Electric-buffer-menu-mode emms-browser-mode emms-mark-mode emms-metaplaylist-mode emms-playlist-mode ert-results-mode etags-select-mode fj-mode gc-issues-mode gdb-breakpoints-mode gdb-disassembly-mode gdb-frames-mode gdb-locals-mode gdb-memory-mode gdb-registers-mode gdb-threads-mode gist-list-mode gnus-article-mode gnus-browse-mode gnus-group-mode gnus-server-mode gnus-summary-mode google-maps-static-mode ibuffer-mode jde-javadoc-checker-report-mode magit-commit-mode magit-diff-mode magit-key-mode magit-log-mode magit-mode magit-reflog-mode magit-show-branches-mode magit-stash-mode magit-status-mode magit-wazzup-mode mh-folder-mode monky-mode notmuch-hello-mode notmuch-search-mode notmuch-show-mode occur-mode org-agenda-mode proced-mode rcirc-mode rebase-mode recentf-dialog-mode reftex-select-bib-mode reftex-toc-mode sldb-mode slime-inspector-mode slime-thread-control-mode slime-xref-mode sr-buttons-mode sr-mode sr-tree-mode sr-virtual-mode tar-mode tetris-mode tla-annotate-mode tla-archive-list-mode tla-bconfig-mode tla-bookmarks-mode tla-branch-list-mode tla-browse-mode tla-category-list-mode tla-changelog-mode tla-follow-symlinks-mode tla-inventory-file-mode tla-inventory-mode tla-lint-mode tla-logs-mode tla-revision-list-mode tla-revlog-mode tla-tree-lint-mode tla-version-list-mode twittering-mode urlview-mode vc-annotate-mode vc-dir-mode vc-git-log-view-mode vc-svn-log-view-mode vm-mode vm-summary-mode w3m-mode wab-compilation-mode xgit-annotate-mode xgit-changelog-mode xgit-diff-mode xgit-revlog-mode xhg-annotate-mode xhg-log-mode xhg-mode xhg-mq-mode xhg-mq-sub-mode xhg-status-extra-mode compilation-mode)))
 '(evil-ex-interactive-search-highlight (quote selected-window))
 '(evil-motion-state-modes (quote (apropos-mode Buffer-menu-mode calendar-mode color-theme-mode command-history-mode dictionary-mode Info-mode speedbar-mode undo-tree-visualizer-mode view-mode)))
 '(evil-move-cursor-back t)
 '(evil-want-visual-char-semi-exclusive nil)
 '(exec-path (quote ("/opt/local/lib/mysql5/bin" "/opt/local/bin" "/usr/local/bin" "/usr/bin" "/bin" "/usr/sbin" "/sbin" "/Applications/Emacs.app/Contents/MacOS/bin")))
 '(fill-column 80)
 '(fold-dwim-mode t)
 '(gud-gdb-command-name "gdb --annotate=1")
 '(hl-line-sticky-flag nil)
 '(ido-completion-buffer "*Ido Completions*")
 '(ido-completion-buffer-all-completions t)
 '(ido-enable-flex-matching t)
 '(ido-everywhere t)
 '(indent-tabs-mode nil)
 '(indicate-empty-lines t)
 '(inferior-lisp-program "/opt/local/bin/sbcl")
 '(inhibit-startup-screen t)
 '(ispell-program-name "/usr/local/bin/aspell")
 '(jslint-shell-command "")
 '(large-file-warning-threshold nil)
 '(ldap-host-parameters-alist (quote (("operator.tamu.edu" base "ou=people,dc=tamu,dc=edu" auth simple))))
 '(menu-bar-mode nil)
 '(message-log-max 5000)
 '(message-send-mail-function (quote message-send-mail-with-sendmail))
 '(minimap-dedicated-window nil)
 '(mmm-submode-decoration-level 1)
 '(mouse-avoidance-mode (quote banish) nil (avoid))
 '(mouse-highlight nil)
 '(mysql2sqlite-mysqldump-database "")
 '(mysql2sqlite-mysqldump-executable "/opt/local/lib/mysql5/bin/mysqldump")
 '(mysql2sqlite-mysqldump-user "root")
 '(mysql2sqlite-sqlite-executable "/opt/local/bin/sqlite3")
 '(ns-alternate-modifier (quote meta))
 '(ns-antialias-text nil)
 '(ns-command-modifier (quote super))
 '(ns-control-modifier (quote control))
 '(ns-function-modifier (quote none))
 '(ns-pop-up-frames nil)
 '(ns-use-qd-smoothing nil)
 '(org-agenda-compact-blocks nil)
 '(org-agenda-restore-windows-after-quit t)
 '(org-agenda-show-all-dates nil)
 '(org-agenda-start-on-weekday nil)
 '(org-agenda-window-setup (quote current-window))
 '(org-link-frame-setup (quote ((vm . vm-visit-folder-other-frame) (gnus . gnus-other-frame) (file . find-file))))
 '(org-log-done (quote note))
 '(org-log-into-drawer t)
 '(org-timeline-show-empty-dates nil)
 '(org-todo-keywords (quote ((sequence "TODO(t)" "IN PROGRESS(i!)" "|" "DONE(d@/!)") (sequence "ON HOLD(h!)" "|" "CANCELLED(c@/!)"))))
 '(package-archives (quote (("gnu" . "http://elpa.gnu.org/packages/") ("marmalade" . "http://marmalade-repo.org/packages/") ("melpa" . "http://melpa.milkbox.net/packages/"))))
 '(php-auto-fill t)
 '(php-blank-line-at-end-of-class t)
 '(php-doc-default-author (quote ("Brian Zwahr" . "bzwahr@tamu.edu")))
 '(php-doc-default-category "Webapp")
 '(php-doc-default-copyright "2012 Texas A&M University")
 '(php-doc-default-license "http://www.gnu.org/licenses/gpl-3.0.html  GPLv3")
 '(php-doc-default-link "http://www.tamu.edu")
 '(php-doc-default-php-version "5.3")
 '(php-doc-default-version "SVN: $Id: $")
 '(php-format-align-array-double-arrows nil)
 '(php-format-break-all-method-call-arguments t)
 '(php-format-break-all-method-chain-links t)
 '(php-hide-show-hide-doc-blocks t)
 '(php-hide-show-ignore-extensions (quote (".phtml")))
 '(php-manual-path "/Users/bzwahr/php-chunked-xhtml")
 '(php-parse-send-to-front (quote (("__construct" nil method public) ("init" nil method public) ("setUp" nil method public) ("tearDown" nil property public))))
 '(php-project-list (quote (("biosafety-occhealth" "/Volumes/webapp-dev/biosafety/occhealth/" "/Users/bzwahr/.ctags/biosafety-occhealth" ("/Volumes/webapp-dev/frameworks/zend-1.11/Zend/" "/usr/local/pear/share/pear/PHPUnit/") "/Volumes/webapp-dev/biosafety/occhealth/tests/phpunit.xml" ("/Volumes/webapp-dev/biosafety/occhealth/tests/application/" "/Volumes/webapp-dev/biosafety/occhealth/library/Biosafety/" "/Volumes/webapp-dev/biosafety/occhealth/application/") (("" . "") "" "" "" "" "" "" "Biosafety" "") "" "Default") ("scheduler" "/Volumes/webapp-dev/scheduler/trunk/" "/Users/bzwahr/.ctags/scheduler" ("/Volumes/webapp-dev/frameworks/zend-1.11/Zend/" "/usr/local/pear/share/pear/PHPUnit/") "/Volumes/webapp-dev/scheduler/trunk/tests/phpunit.xml" ("/Volumes/webapp-dev/scheduler/trunk/library/Scheduler/" "/Volumes/webapp-dev/scheduler/trunk/application/" "/Volumes/webapp-dev/scheduler/trunk/tests/application/") (("" . "") "" "" "" "" "" "" "Scheduler" "") "Default" "") ("biosafety" "/Volumes/webapp-dev/biosafety/trunk/" "/Users/bzwahr/.ctags/biosafety" ("/Volumes/webapp-dev/frameworks/zend-1.11/Zend/" "/usr/local/pear/share/pear/PHPUnit/") "/Volumes/webapp-dev/biosafety/trunk/tests/phpunit.xml" ("/Volumes/webapp-dev/biosafety/trunk/application/" "/Volumes/webapp-dev/biosafety/trunk/library/Biosafety/" "/Volumes/webapp-dev/biosafety/trunk/tests/application") (("" . "") "" "" "" "" "" "" "Biosafety" "") "Default" ""))))
 '(php-tag-arguments (quote ("--PHP-kinds=+cf" "--regex-PHP='/abstract class ([^ ]*)//c/'" "--regex-PHP='/interface ([^ ]*)//c/'" "--regex-PHP='/(public |static |abstract |protected |private )+function ([^ (]*)//f/'")))
 '(php-tag-shell-command "/opt/local/bin/ctags")
 '(php-tags-relative t)
 '(php-test-ask-save nil)
 '(php-test-compile-tests (quote (lint phpcs phpmd)))
 '(php-test-file-extensions (quote ("php" "inc" "phtml")))
 '(php-test-show-command nil)
 '(phpcs-shell-command "/usr/local/pear/bin/phpcs")
 '(phpcs-standard "ZFMode")
 '(phpmd-rulesets (quote (codesize design naming unusedcode)))
 '(phpmd-shell-command "/usr/local/pear/bin/phpmd")
 '(phpunit-shell-command "/usr/local/pear/bin/phpunit")
 '(rmail-preserve-inbox t)
 '(rudel-configured-sessions nil)
 '(scroll-conservatively 0)
 '(send-mail-function (quote sendmail-send-it))
 '(smex-save-file "~/.smex")
 '(solarized-termcolors 256)
 '(speedbar-frame-parameters (quote ((minibuffer) (width . 120) (border-width . 0) (menu-bar-lines . 0) (tool-bar-lines . 0) (unsplittable . t) (left-fringe . 0))))
 '(speedbar-show-unknown-files t)
 '(sql-mysql-program "/opt/local/lib/mysql5/bin/mysql")
 '(switch-buffer-function-list nil)
 '(tab-always-indent (quote complete))
 '(tab-width 4)
 '(tag-shell-command "/usr/local/bin/ctags")
 '(term-scroll-show-maximum-output nil)
 '(term-scroll-to-bottom-on-output t)
 '(trash-directory "~/.Trash")
 '(truncate-partial-width-windows nil)
 '(uniquify-buffer-name-style (quote post-forward-angle-brackets) nil (uniquify))
 '(user-mail-address "bzwahr@tamu.edu")
 '(winner-mode t nil (winner))
 '(woman-use-own-frame nil)
 '(zf-mode-company t)
 '(zf-mode-css-compile-on-save t)
 '(zf-mode-delete-trailing-whitespace t)
 '(zf-mode-hideshow t)
 '(zf-mode-hideshow-hide-on-load t)
 '(zf-mode-hideshowvis t)
 '(zf-mode-js-compile-on-save t)
 '(zf-mode-php-compile-on-save t)
 '(zf-mode-php-electric t)
 '(zf-mode-show-project-in-modeline t)
 '(zf-mode-show-trailing-whitespace t)
 '(zf-mode-use-hyphens-in-viewscript-urls t)
 '(zf-mode-wrap-region t))

(custom-set-faces
 ;; custom-set-faces was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 '(hs-face ((t (:background "#ff8" :foreground "black" :box 1))) t)
 '(mmm-default-submode-face ((t (:background "gray25"))) t)
 '(rst-level-1-face ((t (:background "grey85" :foreground "black"))) t)
 '(rst-level-2-face ((t (:background "grey78" :foreground "black"))) t)
 '(rst-level-3-face ((t (:background "grey71" :foreground "black"))) t)
 '(rst-level-4-face ((t (:background "grey64" :foreground "black"))) t)
 '(rst-level-5-face ((t (:background "grey57" :foreground "black"))) t)
 '(rst-level-6-face ((t (:background "grey50" :foreground "black"))) t)
 '(window-number-face ((nil (:background "black" :foreground "red")))))

 
